Classes/Player/PlayerController.swift

```diff
@@ -53,12 +53,18 @@ class PlayerController: NSObject, Player {
         return self.currentPlayer.view
     }
     
-    public let sessionId = UUID()
+    public var sessionId: String {
+        return self.sessionUUID.uuidString + ":" + (self.mediaSessionUUID?.uuidString ?? "")
+    }
+    
+    let sessionUUID = UUID()
+    var mediaSessionUUID: UUID?
     
     public override init() {
         self.currentPlayer = AVPlayerEngine()
-        self.contentRequestAdapter = KalturaPlaybackRequestAdapter(playSessionId: self.sessionId)
         super.init()
+        // initial creation of play session id adapter
+        self.contentRequestAdapter = KalturaPlaybackRequestAdapter(sessionId: self.sessionId)
```

Comment:
yea but then I have to make the mediaSessionId Internal to initialize it from the loader it doesn't make sense to have a property on player and initialize some of them from the loader seems wrong...